src/events/winston.js

```diff
@@ -0,0 +1,56 @@
+const tracer = require('../tracer.js');
+const moduleUtils = require('./module_utils.js');
+
+/**
+ * Wrap bunyan logs
+ * @param {Function} wrappedFunction The function to wrap from bunyan
+ * @returns {function} emit wrapper function
+ */
+function writeWrapper(wrappedFunction) {
+    return function internalWriteWrapper(chunk, encoding, callback) {
+        if (!tracer.isLoggingTracingEnabled()) {
+            return wrappedFunction.apply(this, [chunk, encoding, callback]);
+        }
+        const traceId = tracer.getTraceId();
+        if (!traceId) {
+            return wrappedFunction.apply(this, [chunk, encoding, callback]);
+        }
+
+        if (!chunk || typeof chunk !== 'object') {
```

Comment:
All if statements can be in separate functions 
(you can merge them to one if statements, but it does not necessarily cause maybe it will be less readable so it up to you).